tests/testthat/test-calculateNotApplicableStatus.R

```diff
@@ -0,0 +1,87 @@
+library(testthat)
+
+test_that("Not Applicable status Table Empty", {
+  outputFolder <- tempfile("dqd_")
+  on.exit(unlink(outputFolder, recursive = TRUE))
+
+  results <- executeDqChecks(
+    connectionDetails = connectionDetailsEunomia,
+    cdmDatabaseSchema = cdmDatabaseSchemaEunomia,
+    resultsDatabaseSchema = resultsDatabaseSchemaEunomia,
+    cdmSourceName = "Eunomia",
+    checkNames = c("cdmTable", "cdmField", "measureValueCompleteness"),
+    # Eunomia COST table has misspelled 'REVEUE_CODE_SOURCE_VALUE'
+    tablesToExclude = c("COST", "CONCEPT", "VOCABULARY", "CONCEPT_ANCESTOR", "CONCEPT_RELATIONSHIP", "CONCEPT_CLASS", "CONCEPT_SYNONYM", "RELATIONSHIP", "DOMAIN"),
+    outputFolder = outputFolder,
+    writeToTable = F
+  )
+
+  # Assumption that Eunomia has empty device_exposure table
+  r <- results$CheckResults[results$CheckResults$checkName == "measureValueCompleteness" &
+                            results$CheckResults$tableName == "device_exposure", ]
+  expect_true(all(r$notApplicable == 1))
+})
+
+test_that("measureConditionEraCompleteness Not Applicable if condition_occurrence empty", {
+  outputFolder <- tempfile("dqd_")
+  on.exit(unlink(outputFolder, recursive = TRUE))
+
+  # Remove records from Condition Occurrence
+  connection <- DatabaseConnector::connect(connectionDetailsEunomia)
```

Comment:
i think it would be safer to create a separate eunomia DB just for these tests. i believe testthat runs the test files in parallel so we might run into weird issues if we're modifying the same DB here.

Reply:
I don't think this is an issue. If I open a Eunomia connection in two R sessions, clearing a table in one session does not affect the other session.

Reply:
@MaximMoinat I think this got lost in the shuffle - let me know what you think and if you're OK creating new connection details for these tests. Thanks!